File: server/apps/api-service/src/main/java/com/sharetimer/apiservice/application/port/in/TimerUseCase.java
package com.sharetimer.apiservice.application.port.in;

import com.sharetimer.apiservice.adapter.in.web.dto.TimerCreateRes;
import com.sharetimer.apiservice.adapter.in.web.dto.TimerInfoRes;
import com.sharetimer.apiservice.application.port.in.command.AddTimestampCommand;
import com.sharetimer.apiservice.application.port.in.command.CreateTimerCommand;
import com.sharetimer.apiservice.application.port.in.command.DeleteTimerCommand;
import com.sharetimer.apiservice.application.port.in.command.GetTimerCommand;
import com.sharetimer.apiservice.application.port.in.command.UpdateTimerCommand;

public interface TimerUseCase {

  TimerCreateRes createTimer(CreateTimerCommand cmd);

  TimerInfoRes getTimerInfo(GetTimerCommand cmd);

  void deleteTimer(DeleteTimerCommand cmd);

  void updateTimer(UpdateTimerCommand cmd);

  void addTimestamp(AddTimestampCommand cmd);
}

File: server/apps/api-service/src/main/kotlin/com/sharetimer/apiservice/application/port/input/TimerUseCase.kt
package com.sharetimer.apiservice.application.port.input

import com.sharetimer.apiservice.application.port.input.command.AddTimestampCommand
import com.sharetimer.apiservice.application.port.input.command.CreateTimerCommand
import com.sharetimer.apiservice.application.port.input.command.DeleteTimerCommand
import com.sharetimer.apiservice.application.port.input.command.GetTimerQuery
import com.sharetimer.apiservice.application.port.input.command.UpdateTimerCommand
import com.sharetimer.apiservice.domain.model.Timer

interface TimerUseCase {
    fun createTimer(cmd: CreateTimerCommand): Timer

    fun getTimerInfo(query: GetTimerQuery): Timer

    fun deleteTimer(cmd: DeleteTimerCommand)

    fun updateTimer(cmd: UpdateTimerCommand)

    fun addTimestamp(cmd: AddTimestampCommand)
}
